src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newStates = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        const StateBfAdd = { ...(newStates[newStates.length - 1] || state) };
+
+        newStates.push({ ...StateBfAdd, ...action.extraData });
+        break;
+      case 'removeProperties':
+        const stateBfRemove = { ...(newStates[newStates.length - 1] || state) };
```

Comment:
I see you intended to use only `const` and so you need to access the previous state using this index.
You could make it in two other ways that are better for performance:
1. Using `array.at` method:
```
const previousState = newStates.at(-1) ?? state;
```
2. Using another const to store the previous state before:
```
const newStates = [];
const stateHistory = { state };

switch ...
  ...
  const previousState = { ...stateHistory.state };
 ... do the transformations in previousState as needed
 stateHistory.state = { ...previousState };
 break;
```